src/utils/__tests__/spid.test.ts

```diff
@@ -0,0 +1,229 @@
+            <saml:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string">
+                AgID
+            </saml:AttributeValue>
+        </saml:Attribute>
+        <saml:Attribute Name="fiscalNumber" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
+            <saml:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string">
+                TINIT-GDASDV00A01H501J
+            </saml:AttributeValue>
+        </saml:Attribute>
+        <saml:Attribute Name="mobilePhone" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
+            <saml:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string">
+                +393331234567
+            </saml:AttributeValue>
+        </saml:Attribute>
+        <saml:Attribute Name="email" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
+            <saml:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string">
+                spid.tech@agid.gov.it
+            </saml:AttributeValue>
+        </saml:Attribute>
+        <saml:Attribute Name="address" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
+            <saml:AttributeValue xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="xs:string">
+                Via Listz 21 00144 Roma
+            </saml:AttributeValue>
+        </saml:Attribute>
+    </saml:AttributeStatement>
+</saml:Assertion>
+</samlp:Response>`;
+
+const aDOMSamlRequest = new DOMParser().parseFromString(
+  aSAMLRequest,
+  "text/xml"
+);
+const aDOMSamlResponse = new DOMParser().parseFromString(
+  aSAMLResponse,
+  "text/xml"
+);
+
+describe("SPID logs", () => {
+  it("should get SPID request id from request", () => {
+    const requestId = getRequestIDFromRequest(aDOMSamlRequest);
+    expect(requestId).toEqual(some("A-REQUEST-ID"));
+  });
+
+  it("should get SPID request id from response", () => {
+    const requestId = getRequestIDFromResponse(aDOMSamlResponse);
+    expect(requestId).toEqual(some("_2d2a89e99c7583e221b4"));
+  });
+
+  it("should get SPID user's fiscal code from response", () => {
+    const fiscalCode = getFiscalNumberFromPayload(aDOMSamlResponse);
+    expect(fiscalCode).toEqual(some("GDASDV00A01H501J"));
+  });
+
+  it("should enqueue valid payload on SPID request", () => {
+    const mockQueueClient = {
+      sendMessage: jest.fn().mockImplementation(() => Promise.resolve())
+    };
+    makeSpidLogCallback((mockQueueClient as unknown) as QueueClient)(
+      "1.1.1.1",
+      aSAMLRequest,
+      "REQUEST"
+    );
+    expect(mockQueueClient.sendMessage).toHaveBeenCalled();
```

Comment:
Do we know what params are sent to sendMessage?  In that case use `toHaveBeenCalledWith` and check the parameters

Reply:
that would require to much work since we need to encode an object into base64